### PR TITLE
Add new extensions/plugins

### PR DESCRIPTION
**Pull request by TakoB222**
Added support for new extensions/plugins:

- icu_ext
- address_standardizer
- address_standardizer
- address_standardizer_data_us
- bool_plperl 
- hll 
- hstore 
- hypopg 
- ip4r 
- jsonb_plperl 
- orafce 
- pg_similarity
- pg_trgm 
- plperl 
- pltcl 
- postgis
- postgis_raster 
- postgis_tiger_geocoder 
- postgis_topology 
- prefix 
- rdkit 
- tds_fdw 
- unaccent

### CODE REVIEW

File: lib/charms/postgresql_k8s/v0/postgresql.py
Comment (dragomirp) [inline lib/charms/postgresql_k8s/v0/postgresql.py:1]: Changes to the lib should increment LIBPATCH version.

File: lib/charms/postgresql_k8s/v0/postgresql.py [hunk 1]
```diff
@@ -25,6 +25,9 @@
 from ops.model import Relation
 from psycopg2 import sql
 from psycopg2.sql import Composed
+from constants import DEPENDENCY_PLUGINS
```
Comment (dragomirp) [inline lib/charms/postgresql_k8s/v0/postgresql.py:28]: The lib shouldn't import from the charm. Those constants should be moved here and imported from here in the charm.

File: src/charm.py [hunk 2]
```diff
@@ -872,16 +874,40 @@ def enable_disable_extensions(self, database: str = None) -> None:
...
+                        extension,
+                        ext,
+                    )
+        elif not enable and extension in DEPENDENCY_PLUGINS:
+            for n, v in REQUIRED_PLUGINS.items():
+                if extension in v and self.config["plugin_" + n + "_enable"]:
+                    skip = True
+                    logger.exception(
+                        f"{n} require plugin {extension}. Switch off first {n} before disable {extension}"
+                    )
```
Comment (dragomirp) [inline src/charm.py:908]: I don't think we need this loop. The first one will run over all the plugins and verify them anyway.

File: src/charm.py [hunk 3]
```diff
@@ -872,16 +874,40 @@ def enable_disable_extensions(self, database: str = None) -> None:
                 for ext in spi_module:
                     extensions[ext] = enable
                 continue
-            if extension in plugins_exception:
-                extension = plugins_exception[extension]
+            extension = plugins_exception.get(extension, extension)
+            if self._check_extension_dependencies(extension, enable):
+                self.unit.status = BlockedStatus(EXTENSIONS_BLOCKING_MESSAGE)
```
Comment (dragomirp) [inline src/charm.py:879]: We should define a different message constant. Something like `unsatisfied plugin dependencies. Please check the logs`

File: tests/integration/test_plugins.py [hunk 4]
```diff
@@ -110,6 +132,24 @@ async def test_plugins(ops_test: OpsTest) -> None:
...
+        "plugin_pg_similarity_enable": ORAFCE_EXTENSION_STATEMENT,
+        "plugin_prefix_enable": PREFIX_EXTENSION_STATEMENT,
+        "plugin_rdkit_enable": RDKIT_EXTENSION_STATEMENT,
+        "plugin_tds_fdw_enable": TDS_FDW_EXTENSION_STATEMENT,
+        "plugin_icu_ext_enable": ICU_EXT_EXTENSION_STATEMENT,
+        "plugin_pltcl_enable": PLTCL_EXTENSION_STATEMENT,
+        "plugin_address_standardizer_enable": ADDRESS_STANDARDIZER_EXTENSION_STATEMENT,
+        "plugin_address_standardizer_data_us_enable": ADDRESS_STANDARDIZER_DATA_US_EXTENSION_STATEMENT,
+        "plugin_postgis_tiger_geocoder_enable": POSTGIS_TIGER_GEOCODER_EXTENSION_STATEMENT,
+        "plugin_postgis_topology_enable": POSTGIS_TOPOLOGY_STATEMENT,
```
Comment (marceloneppel) [inline tests/integration/test_plugins.py:152]: The test for `plugin_postgis_raster_enable` is missing.

**Review by marceloneppel — APPROVED** [review]
LGTM! Thanks @TakoB222!

**Review by dragomirp — APPROVED** [review]
LGTM. Thanks for the contribution.

File: lib/charms/postgresql_k8s/v0/postgresql.py [hunk 5]
```diff
@@ -289,12 +304,18 @@ def enable_disable_extensions(self, extensions: Dict[str, bool], database: str =
                     cursor.execute("SELECT datname FROM pg_database WHERE NOT datistemplate;")
                     databases = {database[0] for database in cursor.fetchall()}
 
+            orderedExtensions = OrderedDict()
+            for plugin in DEPENDENCY_PLUGINS:
+                orderedExtensions[plugin] = extensions[plugin]
```
Comment (dragomirp) [inline lib/charms/postgresql_k8s/v0/postgresql.py:309]:
```suggestion
                orderedExtensions[plugin] = extensions.get(plugin, False)
```
Creating databases passes only enabled plugins, which causes instability in the integration tests.

**Review by taurus-forever — APPROVED** [review]
LGTM. Ready for merge IMHO. Please apply @dragomirp changes and he will merge it.
Thank you, Artem!!!

**Pull request merged**